apis/tags/v1beta1/tagkey_types.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+
+	// "github.com/GoogleCloudPlatform/k8s-config-connector/apis/common/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var TagsTagKeyGVK = GroupVersion.WithKind("TagsTagKey")
+
+// TagsTagKeySpec defines the desired state of TagsTagKey
+// +kcc:spec:proto=google.cloud.resourcemanager.v3.TagKey
+type TagsTagKeySpec struct {
+	// // Required. Defines the parent path of the resource.
+	// *parent.ProjectAndLocationRef `json:",inline"`
+
+	// The TagsTagKey name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Immutable. The resource name of the TagKey's parent. A TagKey can be
+	//  parented by an Organization or a Project. For a TagKey parented by an
+	//  Organization, its parent must be in the form `organizations/{org_id}`. For
+	//  a TagKey parented by a Project, its parent can be in the form
+	//  `projects/{project_id}` or `projects/{project_number}`.
+	// +kcc:proto:field=google.cloud.resourcemanager.v3.TagKey.parent
+	// +required
+	Parent *string `json:"parent,omitempty"`
```

Comment:
Should we have a ref for this?

Reply:
I mean in addition to the Parent which I think we are stuck with for compatibility reasons?